src/test/run-make-fulldeps/coverage/doctest.rs

```diff
@@ -0,0 +1,66 @@
+//! This test ensures that code from doctests is properly re-mapped.
+//! See <https://github.com/rust-lang/rust/issues/79417> for more info.
+//!
+//! Just some random code:
+//! ```
+//! if true {
+//!     // this is executed!
+//!     assert_eq!(1, 1);
+//! } else {
+//!     // this is not!
+//!     assert_eq!(1, 2);
+//! }
+//! ```
+//!
+//! doctest testing external code:
+//! ```
+//! extern crate doctest_crate;
+//! doctest_crate::fn_run_in_doctests(1);
+//! ```
+//!
+//! doctest returning a result:
+//! ```
+//! #[derive(Debug)]
+//! struct SomeError;
+//! let mut res = Err(SomeError);
+//! if res.is_ok() {
```

Comment:
I just realized, the coverage results in `src/test/run-make-fulldeps/coverage-reports/expected_show_coverage.doctest.txt` don't show any examples of coverage count differences within a line.

I think we need at least one, to make sure the column positions are also computed correctly.

Look, for example, at:

https://github.com/rust-lang/rust/blob/bd2f1cb2785f87177249e2bdb628ed782fcd8def/src/test/run-make-fulldeps/coverage-reports/expected_show_coverage.conditions.txt#L30

```rust

   28|      1|    if countdown > 7 {
   29|      1|        countdown -= 4;
   30|      1|    } else if countdown > 2 {
                            ^0
   31|      0|        if countdown < 1 || countdown > 5 || countdown != 9 {
   32|      0|            countdown = 0;
   33|      0|        }
```

There are other examples, in this sample program and others, showing the column-specific counter position. (It's even more clear when viewing coverage results with `--color`, in a terminal.)

It should be pretty easy to add something similar, and compare the doctest column position to the equivalent in a standard rust source coverage result.

Reply:
As the PR description says, the columns are not yet correct. I think a solution for that could be a lot more complex, as rustdoc can have non-uniform column offsets, such as:

````
/// ```
/// some_normal_code();
/// # executed_but_hidden_code();
/// ```
````

In the intermediate generated code, these have the same (no) indentation.

Another interesting case is that you can mix `///`-style comments, and doc attributes, with a lot more complex cases that I actually first read about here: https://blog.guillaume-gomez.fr/articles/2020-11-11+New+doc+comment+handling+in+rustdoc

So in those cases, the column offsets can differ for each line.

Reply:
Ah. I miss that. That's fine then. It certainly a big improvement with the corrected line numbers and we can save the column positions for a future PR. Thanks!